radiation/python/radlw/stencils_gt4py.py

```diff
@@ -0,0 +1,4975 @@
+                                # workaround since max doesn't work in for loop in if statement
+                                tauice[0, 0, 0][ib4] = tmp if tmp > 0.0 else 0.0
+
+                        elif ilwcice == 3:
+                            dgeice = max(5.0, 1.0315 * refice)  # v4.71 value
+                            factor = (dgeice - 2.0) / 3.0
+                            index = max(1, min(45, factor)) - 1
+                            fint = factor - (index + 1)
+
+                            for ib5 in range(nbands):
+                                tmp = cldice * (
+                                    absice3[0, 0, 0][index, ib5]
+                                    + fint
+                                    * (
+                                        absice3[0, 0, 0][index + 1, ib5]
+                                        - absice3[0, 0, 0][index, ib5]
+                                    )
+                                )
+                                # workaround since max doesn't work in for loop in if statement
+                                tauice[0, 0, 0][ib5] = tmp if tmp > 0.0 else 0.0
+
+                    for ib6 in range(nbands):
+                        taucld[0, 0, 0][ib6] = (
+                            tauice[0, 0, 0][ib6]
+                            + tauliq[0, 0, 0][ib6]
+                            + tauran
+                            + tausnw
+                        )
+
+            else:
+                if cfrac > cldmin:
+                    for ib7 in range(nbands):
+                        taucld[0, 0, 0][ib7] = cdat1
+
+            if isubclw > 0:
+                if cfrac < cldmin:
+                    cldf = 0.0
+                else:
+                    cldf = cfrac
+
+    # This section builds mcica_subcol from the fortran into cldprop.
+    # Here I've read in the generated random numbers until we figure out
+    # what to do with them. This will definitely need to change in future.
+    # Only the iovrlw = 1 option is ported from Fortran
+    with computation(PARALLEL), interval(2, None):
+        if lcf1:
+            tem1 = 1.0 - cldf[0, 0, -1]
+
+            for n in range(ngptlw):
+                if cdfunc[0, 0, -1][n] > tem1:
+                    cdfunc[0, 0, 0][n] = cdfunc[0, 0, -1][n]
+                else:
+                    cdfunc[0, 0, 0][n] = cdfunc[0, 0, 0][n] * tem1
+
+    with computation(PARALLEL), interval(1, None):
+        if lcf1:
+            tem1 = 1.0 - cldf[0, 0, 0]
+
+            for n2 in range(ngptlw):
+                if cdfunc[0, 0, 0][n2] >= tem1:
+                    lcloudy[0, 0, 0][n2] = 1
+                else:
+                    lcloudy[0, 0, 0][n2] = 0
+
+            for n3 in range(ngptlw):
+                if lcloudy[0, 0, 0][n3] == 1:
+                    cldfmc[0, 0, 0][n3] = 1.0
+                else:
+                    cldfmc[0, 0, 0][n3] = 0.0
+
+            cldtau = taucld[0, 0, 0][6]
+
+
+stpfac = 296.0 / 1013.0
+
```

Comment:
I think you explained for the shortwave adding code to test the smaller units when a flag is turned on. Do you plan to do that in this file as well? If so before or after this PR is merged? I am fine with either way. Alternatively you can have test code in a separate file that sets up variables and calls individual stencils/functions.

Reply:
Will add the flag now